scvi/train/_trainingplans.py

```diff
@@ -209,44 +245,15 @@ def compute_and_log_metrics(
         kl_global = loss_recorder.kl_global
 
         # use the torchmetric object for the ELBO
-        elbo_metric(
-            rec_loss,
-            kl_local,
-            kl_global,
-            n_obs_minibatch,
+        metrics.update(
+            reconstruction_loss=rec_loss,
+            kl_local=kl_local,
+            kl_global=kl_global,
+            n_obs_minibatch=n_obs_minibatch,
         )
-        # e.g., train or val mode
-        mode = elbo_metric.mode
         # pytorch lightning handles everything with the torchmetric object
-        self.log(
-            f"elbo_{mode}",
-            elbo_metric,
-            on_step=False,
-            on_epoch=True,
-            batch_size=n_obs_minibatch,
-        )
-
-        # log elbo components
-        self.log(
-            f"reconstruction_loss_{mode}",
-            rec_loss / elbo_metric.n_obs_total,
-            reduce_fx=torch.sum,
-            on_step=False,
-            on_epoch=True,
-            batch_size=n_obs_minibatch,
-        )
-        self.log(
-            f"kl_local_{mode}",
-            kl_local / elbo_metric.n_obs_total,
-            reduce_fx=torch.sum,
-            on_step=False,
-            on_epoch=True,
-            batch_size=n_obs_minibatch,
-        )
-        # default aggregation is mean
-        self.log(
-            f"kl_global_{mode}",
-            kl_global,
+        self.log_dict(
```

Comment:
One naive question as this is not fresh in my mind anymore. Where are the metrics explicitly reset to 0?

Reply:
They get set to 0 on __init__ https://github.com/scverse/scvi-tools/pull/1529/files/2eb5cb855bfaa3fc2a7a45f057928fe1742dabc6#diff-8fc39d9964444c66caf3545a773bef7402a7a73386aa2761f06930d7759353daR135 as well as when `n_obs_training` is set, which happens in https://github.com/scverse/scvi-tools/blob/master/scvi/train/_trainrunner.py#L70.